src/utils/get-server-status.ts

```diff
@@ -0,0 +1,181 @@
+import console, { log } from 'console'
+
+const SERVER_STATUSES: ServerStatus[] = []
+
+// Function to calculate successes and failures, either total or consecutives.
+// If consecutive success/failure equals to threshold, it will send notifications
+const calculateSuccessFailure = ({
+  errorName,
+  probeStatus,
+  validation,
+  threshold,
+}: {
+  errorName: string
+  probeStatus: Status
+  validation: ValidateResponseStatus
+  threshold: number
+}) => {
+  log(`Checking for ${errorName} in probe...`)
+  switch (probeStatus.isDown) {
+    case false:
+      // If current probe is UP and trigger equals true
+      if (validation.status) {
+        // Add failure count
+        probeStatus.shouldSendNotification = false
+        probeStatus.consecutiveSuccesses = 0
+        probeStatus.totalFailures += 1
+        probeStatus.consecutiveFailures += 1
+
+        // If count is larger or equal than threshold...
+        // Set status to down
+        if (probeStatus.consecutiveFailures === threshold) {
+          log(`Send error for ${errorName}`)
+          probeStatus.shouldSendNotification = true
+          probeStatus.isDown = true
+        }
+      } else {
+        // Server is doing fine
+        probeStatus.shouldSendNotification = false
+        probeStatus.consecutiveFailures = 0
+        probeStatus.consecutiveSuccesses += 1
+        probeStatus.totalSuccesses += 1
+      }
+
+      return probeStatus
+    case true:
+      // If current probe is DOWN and trigger equals true
+      if (validation.status) {
+        // Server is not doing fine
+        probeStatus.shouldSendNotification = false
+        probeStatus.consecutiveSuccesses = 0
+        probeStatus.consecutiveFailures += 1
+        probeStatus.totalFailures += 1
+      } else {
+        // Add success count
+        probeStatus.shouldSendNotification = false
+        probeStatus.consecutiveFailures = 0
+        probeStatus.totalSuccesses += 1
+        probeStatus.consecutiveSuccesses += 1
+
+        // If count is larger or equal than threshold...
+        // Set status to UP
+        if (probeStatus.consecutiveSuccesses === threshold) {
+          log(`Send resolved for ${errorName}`)
+          probeStatus.shouldSendNotification = true
+          probeStatus.isDown = false
+        }
+      }
+
+      return probeStatus
+    default:
+      return probeStatus
+  }
+}
+
+export const getServerStatus = ({
+  probe,
+  validatedResp,
+  threshold,
+}: {
+  config: Config
+  probe: Probe
+  validatedResp: ValidateResponseStatus[]
+  threshold: number
+}) => {
+  try {
+    // Get Probe ID and Name
+    const { id, name } = probe
+    const results: Array<Status> = []
+
+    // Initialize server status
+    // This checks if there are no item in SERVER_STATUSES
+    // that doesn't have item with ID === id, push new ServerStatus
+    if (!SERVER_STATUSES.find((item) => item.id === id)) {
+      SERVER_STATUSES.push({
+        id,
+        name,
+        requests: 0,
```

Comment:
can we put this static json element as constant?
example: `INIT_SERVER_STATUS`
So we can do concatenation such as:
```
SERVER_STATUSES.push ({
  id,
  name,
  ...INIT_SERVER_STATUS
})
```

Reply:
Sure thing